File: app/app/src/main/java/com/solutions/note_it/data/Note.kt
package com.solutions.note_it.data

import com.google.gson.annotations.SerializedName

data class Note(
    @field:SerializedName("id") val id: String = "",
    @field:SerializedName("title") val title: String = "",
    @field:SerializedName("summary") val summary: String = "",
    @field:SerializedName("content") val content: String = "",
    @field:SerializedName("category") val category: Category = Category(type = CategoryType.Note.ordinal)
) {
}

File: api/structs/api/note.go
package api

import "time"

type Note struct {
	Id         string    `json:"id"`
	CreatedAt  time.Time `json:"createdAt"`
	ModifiedAt time.Time `json:"modifiedAt"`
	Title      string    `json:"title"`
	Summary    string    `json:"summary"`
	Content    string    `json:"content"`
	Category   Category  `json:"category"`
}
